bigframes/core/compile/scalar_op_compiler.py

```diff
@@ -656,6 +670,84 @@ def isin_op_impl(x: ibis_types.Value, op: ops.IsInOp):
+        # UTC str to be properly handled.
+        x = (
+            ibis.case()
+            .when(
+                x.re_search(TIMEZONE_POS_REGEX),
+                (
+                    (
+                        x.substr(0, x.length() - 6).to_timestamp(op.format)
+                        if op.format
+                        else x.substr(0, x.length() - 6)
+                    )
+                    .cast(ibis_dtypes.Timestamp(timezone="UTC"))
+                    .cast(ibis_dtypes.int64)
+                    - x.substr(x.length() - 5, 2).cast(ibis_dtypes.int64)
+                    * UNIT_TO_US_CONVERSION_FACTORS["h"]
+                    - x.substr(x.length() - 2, 2).cast(ibis_dtypes.int64)
+                    * UNIT_TO_US_CONVERSION_FACTORS["m"]
+                )
+                .to_timestamp(unit="us")
+                .cast(ibis_dtypes.Timestamp(timezone="UTC")),
+            )
+            .when(
+                x.re_search(TIMEZONE_NEG_REGEX),
+                (
+                    (
+                        x.substr(0, x.length() - 6).to_timestamp(op.format)
+                        if op.format
+                        else x.substr(0, x.length() - 6)
+                    )
+                    .cast(ibis_dtypes.Timestamp(timezone="UTC"))
+                    .cast(ibis_dtypes.int64)
+                    + x.substr(x.length() - 5, 2).cast(ibis_dtypes.int64)
+                    * UNIT_TO_US_CONVERSION_FACTORS["h"]
+                    + x.substr(x.length() - 2, 2).cast(ibis_dtypes.int64)
+                    * UNIT_TO_US_CONVERSION_FACTORS["m"]
+                )
+                .to_timestamp(unit="us")
+                .cast(ibis_dtypes.Timestamp(timezone="UTC")),
+            )
+            .when(
+                x.re_search(UTC_REGEX),
+                (
+                    x.substr(0, x.length() - 1).to_timestamp(op.format)
+                    if op.format
+                    else x.substr(0, x.length() - 1)
+                ).cast(ibis_dtypes.Timestamp(timezone="UTC")),
+            )
+            .else_(
+                (x.to_timestamp(op.format) if op.format else x).cast(
+                    ibis_dtypes.Timestamp(timezone="UTC")
+                )
+            )
+            .end()
+        )
+    elif x.type() == ibis_dtypes.Timestamp(timezone="UTC"):
+        return x
+    elif x.type() != ibis_dtypes.timestamp:
+        unit = op.unit if op.unit is not None else "ns"
+        if unit not in UNIT_TO_US_CONVERSION_FACTORS:
+            raise ValueError(f"Cannot convert input with unit '{unit}'.")
+        x_converted = x * UNIT_TO_US_CONVERSION_FACTORS[unit]
+        x_converted = x_converted.cast(ibis_dtypes.int64)
+        # Note: Due to an issue where casting directly to a non-UTC
+        # timezone does not work, we first cast to UTC. This seems
+        # to bypass a potential bug in Ibis's cast function, allowing
+        # for subsequent casting to a non-UTC timezone. Further
```

Comment:
Sorry about the confusion, will update the comment, this means without timezone.  Basically this is for utc=True vs utc=False. Because of some unknown issue related to data type, potentially because of ibis, it's impossible to cast to the proper type when utc=False, unless cast it to utc timezone first.

Reply:
Based on tests it would appear although the result of int64 to_timestamp is in utc timezone, the cast function think the datatype is actually without timezone, and skip the cast, this is to fix the issue.

Reply:
Comment updated.